src/Geno/C++/Application.cpp

```diff
@@ -141,13 +141,13 @@ void Application::HandleCommandLineArgs( int NumArgs, char** ppArgs )
 		case 1:
 		{
 			m_ExePath = std::filesystem::absolute( ppArgs[ 0 ] ).lexically_normal();
-#if defined( _WIN32 ) || defined( __linux__ )
+#if defined( _WIN32 ) || defined( __linux__ ) || defined( __APPLE__ )
 			m_AppDir  = m_ExePath.parent_path();
 			m_DataDir = m_AppDir.parent_path().parent_path() / L"data";
-#elif defined( __APPLE__ ) // _WIN32 || __linux__
+#elif defined( BUILD_XCODE ) // _WIN32 || __linux__ || __APPLE__
```

Comment:
I don't think this block is ever going to get compiled. Since we are now using __APPLE__ above.

Reply:
Additionally, isn't there a built in define for detecting that we are compiling with Xcode?

Reply:
> Additionally, isn't there a built in define for detecting that we are compiling with Xcode?

Not afaik, I tried searching for it but nothing came up.

Reply:
> I don't think this block is ever going to get compiled. Since we are now using **APPLE** above.

Right ;)